### PR TITLE
Scene Objects for None Context

### PR DESCRIPTION
**Pull request by Licini**
Allowing registering scene objects for `None` context, so the users can construct scenes in pure terminal environment without any visualizers. (also needed for writing tests on our base scene objects)
Changes are:
- allowing `detect_current_context` to return None instead of raising an error.
(also NoSceneObjectContextError is no longer needed because of this change)
- Registering the base scene objects classes at `compas.scene` for context `None`
- Adding not-implemented `draw` functions to base classes for abstract method check.
- Removing `abstractmethod` decorators on base classes so they can still be created without further implementation.
- Although can be created and serialised without context `Scene.redraw()` will still raise a no context errror, since it does not make sense to be called without one.

Testing script (works directly in terminal)
```python
from compas.geometry import Box
from compas.datastructures import Mesh
from compas.datastructures import Network
from compas.datastructures import VolMesh


from compas.scene import Scene
from compas.data import json_dump
import compas


scene = Scene()

box = Box.from_width_height_depth(1, 1, 1)
mesh = Mesh.from_polyhedron(6)
network = Network.from_obj(compas.get('lines.obj'))
volmesh = VolMesh.from_meshgrid(2, 2, 2)

scene.add(box)
scene.add(mesh)
scene.add(network)
scene.add(volmesh)

scene.print_hierarchy()

# scene.redraw() # this is still designed to raise a no context error

json = json_dump(scene, "temp/scene.json")
```
output:
```
└── <TreeNode root>
    ├── <TreeNode Box>
    ├── <TreeNode Mesh>
    ├── <TreeNode Network>
    └── <TreeNode VolMesh>
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -42,10 +42,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * Fixed typo in name `Rhino.Geometry.MeshingParameters` in `compas_rhino.geometry.RhinoBrep.to_meshes()`.
 * Fixed `TypeErrorException` when serializing a `Mesh` which has been converted from Rhino.
 * Fixed color conversions in `compas_rhion.conversions.mesh_to_compas`.
+* Allowed `SceneObject` registration for `None` context.
```
Comment (tomvanmele) [inline CHANGELOG.md]: would prefer: "Changed SceneObject registration to allow for...

**Comment by tomvanmele** [comment]
@Licini how is this going?

**Comment by Licini** [comment]
@tomvanmele This is ready to be reviewed again. Just brought update with main.

**Review by tomvanmele — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (Licini) [reply CHANGELOG.md]: Sounds good, done and merging!

**Pull request merged**